student/tests.py

```diff
@@ -1,15 +1,84 @@
 import pytest
+from django.contrib.auth.models import User
+from django.db.models import QuerySet
 
+from student.models import Student, EducationalInstitution
+from datetime import date
 
-def test_basic_math_operation():
-    assert 10 == 2 * 5
+
+@pytest.fixture
+def example_institution_a():
+    institution = EducationalInstitution(name="A")
+    institution.save()
+    return institution
+
+
+@pytest.fixture
+def example_institution_b():
+    institution = EducationalInstitution(name="B")
+    institution.save()
+    return institution
+
+
+@pytest.fixture
+def example_student_a(example_institution_a):
+    example_user = User.objects.create_user("test", "testpassword")
+    example_user.save()
+    student = Student(user=example_user, full_name="test user", email="test@test.com", date_of_birth=date(2020, 7, 13),
+                      phone_number="0000000000", educational_institution=example_institution_a,
+                      graduation_date=date(2020, 7, 13))
+    student.save()
+    return student
 
 
 @pytest.fixture
-def numbers_list():
-    return [1, 2, 3, 4]
+def example_student_b(example_institution_b):
+    example_user = User.objects.create_user("test2", "testpassword")
+    example_user.save()
+    student = Student(user=example_user, full_name="test user2", email="test2@test.com",
+                      date_of_birth=date(2023, 7, 13),
+                      phone_number="0000000000", educational_institution=example_institution_b,
+                      graduation_date=date(2023, 7, 13))
+    student.save()
+    return student
+
+
+@pytest.mark.django_db
+def test_get_institutions_with_enrolled_students_returns_institutions_with_enrolled_students_as_query_set(
+        example_student_a, example_institution_a, example_institution_b):
+    institution_query_set = EducationalInstitution.get_institutions_with_enrolled_students()
+    assert isinstance(institution_query_set, QuerySet)
+    assert all(isinstance(inst, EducationalInstitution) for inst in institution_query_set)
+    assert list(institution_query_set.values_list("name")) == [(example_institution_a.name,)]
```

Comment:
I don't understand this, on line 49 you get a list of the institutions and here you compare it to the `example_institution_a,name` that you put in a list?

Reply:
I'm checking that the whole list that I got back from the function contains only the institutions it should, comparing the whole list makes sure there aren't any objects in the list that shouldn't be there.
I'm comparing the name field because this is how you guys showed us to do it, instead of comparing the institution objects directly.